python/ql/test/library-tests/PointsTo/new/code/w_function_values.py

```diff
@@ -0,0 +1,32 @@
+def test_conditoinal_function(cond):
+    def foo():
+        return "foo"
+
+    def bar():
+        return "bar"
+
+    if cond:
+        f = foo
+    else:
+        f = bar
+
+    sink = f()
+    return sink
+
+
+f_false = test_conditoinal_function(False)
+f_true = test_conditoinal_function(True)
+
+
+def foo():
+    return "foo"
+
+
+def test_redefinition():
+    f = foo
+
+    def foo():
+        return "refined"
+
+    sink = f()
+    return sink
```

Comment:
Wouldn't we need to actually call the function? (as you do with test_conditoinal_function)

```suggestion
def test_redefinition():
    def foo():
        return "foo"
        
    f = foo

    def foo():
        return "foo redefined"

    sink_foo = f()
    sink_redefined = foo()
    return sink_foo, sink_redefined
    
foo, redefined = test_redefinition()
assert foo == "foo"
assert redefined == "foo redefined"
```

I'm a little in doubt about the part where we return both function results, but thinking about it now, I think it's rather nice -- it does mean we're testing **both** our ability to unpack multiple return results _and_ the function redefinition, which is not really what we want I guess. We can leave it out if you don't like it.

Reply:
Ah, it was just my linter complaining. Making it less likely to happen in real code but perfectly fine for a test case.